src/modules/messagePreview/module.js

```diff
@@ -25,27 +30,43 @@ subscriptions.set('messageCreate', async (client, message) => {
         if (
             !msg.channel
                 .permissionsFor(message.author)
-                ?.has([Permissions.FLAGS.VIEW_CHANNEL, Permissions.FLAGS.READ_MESSAGE_HISTORY])
+                ?.has([
+                    PermissionsBitField.Flags.ViewChannel,
+                    PermissionsBitField.Flags.ReadMessageHistory,
+                ])
         )
             return;
         if (!msg.member) await msg.guild.members.fetch(msg.author);
 
-        const embed = new MessageEmbed()
-            .setAuthor(msg.member.displayName, msg.member.displayAvatarURL({ dynamic: true }))
-            .setDescription(msg.content)
+        const embed = new EmbedBuilder()
+            .setAuthor({
+                name: msg.member.displayName,
+                iconURL: msg.member.displayAvatarURL(),
+                url: msg.url,
+            })
             .setColor(msg.member.displayColor)
-            .setFooter(`Requested by ${message.member.displayName}`)
-            .setTimestamp(msg.createdTimestamp);
+            .setFooter({ text: `Requested by ${message.member.displayName}` })
+            .setTimestamp(msg.createdTimestamp)
+            .setURL(msg.url);
 
+        const additionalEmbeds = [];
+        if (msg.content) embed.setDescription(msg.content);
         if (msg.attachments.size) {
-            embed.setImage(msg.attachments.first().url);
-            embed.addField(
-                'Attachments:',
-                msg.attachments.map((a) => `[${a.name}](${a.url})`).join('\n')
-            );
+            const imgs = msg.attachments.filter((a) => a.contentType.startsWith('image/'));
+            Array.from(imgs.entries()).forEach(([, a], i) => {
+                console.log(a, i);
```

Comment:
debug logging. should be removed or converted to a propper message

Reply:
Affirmative